helpers.go

```diff
@@ -26,6 +26,16 @@ type Optional interface {
 	IsDefined() bool
 }
 
+// UnknownsUnmarshaler provides a method to unmarshal unknown struct fileds and save them as you want
+type UnknownsUnmarshaler interface {
+	UnmarshalUnknown(key string, value interface{})
+}
+
+// UnknownsMarshaler provides a method to write additional struct fields
+type UnknownsMarshaler interface {
+	MarshalUnknowns() map[string]interface{}
```

Comment:
I don't particularly like having a map return value.

IMO, it's better to have `MarshalUnknowns(w *jwriter.Writer, empty bool)`. This way, using a map (or even interface-es) will not be required. Also this allows for more use cases.

Reply:
then check the first version of this PR here: https://github.com/mailru/easyjson/pull/267/files